test/topology_experimental_raft/test_topology_upgrade_not_stuck_after_recent_removal.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import logging
+import pytest
+import time
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.rest_client import inject_error
+from test.pylib.util import wait_for_cql_and_get_hosts
+from test.topology.conftest import skip_mode
+from test.topology.util import wait_until_topology_upgrade_finishes, \
+        wait_for_cdc_generations_publishing, \
+        check_system_topology_and_cdc_generations_v3_consistency
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_topology_upgrade_not_stuck_after_recent_removal(request, manager: ManagerClient):
+    """
+    Regression test for https://github.com/scylladb/scylladb/issues/18198.
+    1. Create a two node cluster in legacy mode
+    2. Remove one of the nodes
+    3. Upgrade the cluster to raft topology.
+    4. Verify that the upgrade went OK and it did not get stuck.
+    """
+    # First, force the nodes to start in legacy mode due to the error injection
+    cfg = {'error_injections_at_startup': ['force_gossip_based_join']}
+
+    logging.info("Creating a two node cluster")
+    servers = [await manager.server_add(config=cfg) for _ in range(2)]
+    cql = manager.cql
+    assert(cql)
+
+    srv1, srv2 = servers
+
+    logging.info("Removing the second node in the cluster")
+    await manager.decommission_node(srv2.server_id)
```

Comment:
Commit message says `use nodetool removenode on the second node` --- I guess it doesn't matter which one we use? (But code should be consistent with commit message)

Reply:
Done, updated the commit message